src/Util/PaginationHelper.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Util;
+
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * @author Jonathan Doelfs <jd@sodatech.com>
+ */
+class PaginationHelper
+{
+    private $requestStack;
+    private $resourceMetadataFactory;
+    private $resourceMetadata;
+
+    private $enabled = true;
+    private $clientEnabled = false;
+    private $clientItemsPerPage = false;
+    private $itemsPerPage = 30;
+    private $maximumItemsPerPage = null;
+
+    private $parameterNamePage = 'page';
+    private $parameterNameEnabled = 'pagination';
+    private $parameterNameItemsPerPage = 'itemsPerPage';
+
+    private $resourceClass;
+    private $operationName;
+
+    public function __construct(RequestStack $requestStack, ResourceMetadataFactoryInterface $resourceMetadataFactory, string $resourceClass, string $operationName = null, bool $enabled = true, bool $clientEnabled = false, bool $clientItemsPerPage = false, int $itemsPerPage = 30, string $parameterNamePage = 'page', string $parameterNameEnabled = 'pagination', string $parameterNameItemsPerPage = 'itemsPerPage', int $maximumItemsPerPage = null)
```

Comment:
The goal of this class was to make it as easy as possible to use it for custom data providers. If the enabled, clientEnabled, ... parameters have to be passed to the different methods you also have to inject them to the constructor of your data provider, which makes it more complicated.
I would go with the factory so you just have to inject the pagination factory into your data provider.

Regarding the name and maybe also the namespace, I think we can find something better